lib/saml_idp/request.rb

```diff
@@ -115,10 +115,11 @@ def valid_signature?
     end
 
     def service_provider?
-      service_provider.valid?
+      service_provider && service_provider.valid?
```

Comment:
I don't think `ServiceProvider.valid?` is implemented correctly.   Take a look, it doesn't check any of the attributes at all.  

Firstly I don't think it can ever return false can it?  In which case it should be removed... and we are basically saying that any non-nil SP is valid.

However didn't you just say "ServiceProvider is valid if it has (a) a fingerprint and (b) a metadata_url.".. it should at least check those.  That would fix out bug... since the bug occurs before a all-nil SPs are valid? right now.